src/components/Trade/TradeInput.tsx

```diff
@@ -47,23 +51,102 @@ export const TradeInput = ({ history }: RouterProps) => {
-    name: ['quote', 'buyAsset', 'sellAsset', 'feeAssetFiatRate', 'quoteError'],
-  }) as [TS['quote'], TS['buyAsset'], TS['sellAsset'], TS['feeAssetFiatRate'], TS['quoteError']]
-  const {
-    updateQuote,
-    checkApprovalNeeded,
-    getSendMaxAmount,
-    updateTrade,
-    feeAsset,
-    refreshQuote,
-  } = useSwapper()
+  const [
+    quote,
+    buyTradeAsset,
+    sellTradeAsset,
+    feeAssetFiatRate,
+    quoteError,
+    sellAssetAccount,
+    selectedAssetAccount,
+    action,
+    amount,
+  ] = useWatch({
+    name: [
+      'quote',
+      'buyAsset',
+      'sellAsset',
+      'feeAssetFiatRate',
+      'quoteError',
+      'sellAssetAccount',
+      'selectedAssetAccount',
+      'action',
+      'amount',
+    ],
+  }) as [
+    TS['quote'],
+    TS['buyAsset'],
+    TS['sellAsset'],
+    TS['feeAssetFiatRate'],
+    TS['quoteError'],
+    TS['sellAssetAccount'],
+    TS['selectedAssetAccount'],
+    TS['action'],
+    TS['amount'],
+  ]
+  const { updateQuote, checkApprovalNeeded, getSendMaxAmount, updateTrade, feeAsset } = useSwapper()
   const translate = useTranslate()
   const selectedCurrencyToUsdRate = useAppSelector(selectFiatToUsdRate)
   const {
     state: { wallet },
   } = useWallet()
 
+  const accountIds = useAppSelector(state => {
+    const assetId = sellTradeAsset?.asset?.assetId
+    return assetId ? selectAccountIdsByAssetId(state, { assetId }) : []
+  })
+
+  const shouldShowAccountSelection = sellTradeAsset?.asset && accountIds.length > 1
+
+  const sellAssetId = sellTradeAsset?.asset?.assetId
+  const highestFiatBalanceAccount = useAppSelector(state =>
+    selectHighestFiatBalanceAccountByAssetId(state, {
+      assetId: sellAssetId ?? '',
+    }),
+  )
+
+  useEffect(
+    () => setValue('sellAssetAccount', selectedAssetAccount ?? highestFiatBalanceAccount),
+    [selectedAssetAccount, highestFiatBalanceAccount, setValue, sellTradeAsset, buyTradeAsset],
+  )
+
+  const updateQuoteClosure = useCallback(() => {
+    const sellAssetChainId = sellAssetAccount ? fromAccountId(sellAssetAccount).chainId : undefined
+    if (
+      buyTradeAsset?.asset &&
+      sellTradeAsset?.asset &&
+      sellAssetAccount &&
+      sellAssetChainId === sellTradeAsset.asset.chainId
+    ) {
+      updateQuote({
+        forceQuote: true,
+        amount: amount ?? '0',
+        sellAsset: sellTradeAsset.asset,
+        buyAsset: buyTradeAsset.asset,
+        feeAsset,
+        action: action ?? TradeAmountInputField.SELL,
+        selectedCurrencyToUsdRate,
+      })
+    }
+  }, [
+    action,
+    amount,
+    buyTradeAsset?.asset,
+    feeAsset,
+    selectedCurrencyToUsdRate,
+    sellAssetAccount,
+    sellTradeAsset?.asset,
+    updateQuote,
+  ])
+
+  // Refresh quote
+  useEffect(() => {
+    updateQuoteClosure()
+  }, [updateQuoteClosure])
+
+  // Update the quote every 30 seconds
+  useInterval(async () => await updateQuoteClosure(), 1000 * 30)
```

Comment:
Getting some tsc warnings on those `await updateQuoteClosure()`: `[tsserver 80007] [I] 'await' has no effect on the type of this expression.`
I guess we should be safe not awaiting them?

Reply:
Checks out, after the refactor `updateQuoteClosure` is no longer async.

Good catch, addressed in https://github.com/shapeshift/web/pull/2471/commits/487db79b789cb99b402877f6711a8ac60830943e and revreted incorrectly committed debug value in https://github.com/shapeshift/web/pull/2471/commits/b2211f632ee50f190f7c3b9dc1f6833f3e830297.